### PR TITLE
 AMB-1769-Convert-MESH-CSV-records-to-FHIR-objects

### PR DESCRIPTION
**Pull request by almas-liaqat**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: backend/pyutils/csv_to_model.py [hunk 1]
```diff
@@ -1,42 +1,60 @@
...
+        "unique_id": row.get("UNIQUE_ID", None),
+        "failure_reasons": reasons,
+    }
+
+    immunisation_failure = ImmunizationErrorModel(**failed_record)
+
+    return immunisation_failure
+
+
+def read_csv_to_model(
```
Comment (artronics) [inline backend/pyutils/csv_to_model.py:28]: This function shouldn't expect a file. This code should perform under the assumption that it receives the content as string. You can read a file during test and pass it to this function. If this function assumes file then, we can't use it in the actual implementation because the content comes from S3 not reading a file.

File: lambda_code/tests/test_mesh.py [hunk 2]
```diff
@@ -76,21 +83,69 @@ def setUp(self):
     @mock_s3
     def test_write_to_s3(self):
```
Comment (artronics) [inline lambda_code/tests/test_mesh.py:84]: What's the purpose of this test? If we want to make sure that the content is written back, then we need to mock the `MeshOutputHandler` and assert if the write report has been called.

File: lambda_code/tests/test_mesh.py [hunk 3]
```diff
@@ -76,21 +83,69 @@ def setUp(self):
...
+
+    def test_successful_immunizations_happy_output(self):
+        """Test that the successful immunization record is formatted correctly"""
+        mesh_csv = MeshCsvParser(self.csv_file_path)
+        immunisation_records, _ = mesh_csv.parse()
+        immunisation_record_json = json.dumps(
+            immunisation_records[0].dict(), indent=4, default=str
+        )
+
+        assert json.loads(immunisation_record_json) == json.loads(
```
Comment (artronics) [inline lambda_code/tests/test_mesh.py:141]: Please use `unittest` assertions. They'll give you diff and they're integrated with the test-runner. Like `self.assertDictEqual()` .

File: lambda_code/src/old_csv_to_model.py
Comment (artronics) [inline lambda_code/src/old_csv_to_model.py:1]: Is this file still necessary? It's called old 😄

File: lambda_code/src/mesh.py
Comment (artronics) [inline lambda_code/src/mesh.py:1]: Can you please remove all the `TODO`s if they are not relevant anymore. Only keep the one that still needs addressing. For example you create an Error class which means the one that's in this file is not used anymore.

File: lambda_code/tests/test_mesh.py (hunk 2 above)
Comment (nathanwall) [reply lambda_code/tests/test_mesh.py:84]: I think this one is yours

**Review by artronics — APPROVED** [review]

**Review by artronics — APPROVED** [review]
